example-hca-metal.c

```diff
@@ -253,137 +307,300 @@ int main(int argc, char *argv[]) {
 
     printf("AES - ECB\n");
     oldcount = metal_cpu_get_timer(cpu);
-    metal_sifive_scl.aes_func.setkey(&metal_sifive_scl, SCL_AES_KEY128, key128_2);
-    metal_sifive_scl.aes_func.cipher(&metal_sifive_scl, SCL_AES_ECB, SCL_ENCRYPT, SCL_LITTLE_ENDIAN_MODE, 1, (uint8_t *)plaintext_le, (uint8_t *)tmp);
-    cyclecount = metal_cpu_get_timer(cpu)-oldcount;
-   
+    if (SCL_OK == metal_sifive_scl.aes_func.setkey(&metal_sifive_scl, SCL_AES_KEY128, key128_2))
+    {
+        if (SCL_OK == metal_sifive_scl.aes_func.cipher(&metal_sifive_scl, SCL_AES_ECB, SCL_ENCRYPT, SCL_LITTLE_ENDIAN_MODE, 1, (uint8_t *)plaintext_le, (uint8_t *)tmp))
+        {
```

Comment:
Usually we tend to test error case in order to avoid nested if statement. Is there a reason to do it this way?

Reply:
not really ;)